src/lavinmq/mqtt/client.cr

```diff
@@ -0,0 +1,261 @@
+require "openssl"
+require "socket"
+require "../client"
+require "../error"
+require "../rough_time"
+require "./session"
+require "./protocol"
+
+module LavinMQ
+  module MQTT
+    class Client < LavinMQ::Client
+      include Stats
+      include SortableJSON
+
+      getter channels, log, name, user, client_id, socket, remote_address, connection_info
+      getter? clean_session
+      @connected_at = RoughTime.unix_ms
+      @channels = Hash(UInt16, Client::Channel).new
+      @session : MQTT::Session?
+      rate_stats({"send_oct", "recv_oct"})
+      Log = LavinMQ::Log.for "mqtt.client"
+
+      def vhost
+        @broker.vhost
+      end
+
+      def initialize(@socket : ::IO,
+                     @connection_info : ConnectionInfo,
+                     @user : User,
+                     @broker : MQTT::Broker,
+                     @client_id : String,
+                     @clean_session : Bool = false,
+                     @keepalive : UInt16 = 30,
+                     @will : MQTT::Will? = nil)
+        @io = MQTT::IO.new(@socket)
+        @lock = Mutex.new
+        @waitgroup = WaitGroup.new(1)
+        @remote_address = @connection_info.src
+        @local_address = @connection_info.dst
+        @name = "#{@remote_address} -> #{@local_address}"
+        @metadata = ::Log::Metadata.new(nil, {vhost: @broker.vhost.name, address: @remote_address.to_s, client_id: client_id})
+        @log = Logger.new(Log, @metadata)
+        @broker.vhost.add_connection(self)
+        @log.info { "Connection established for user=#{@user.name}" }
+        spawn read_loop
+      end
+
+      def client_name
+        "mqtt-client"
```

Comment:
fixed? shouldn't include `client_id` or anything?

Reply:
we put this a sfixed for the time being bc. we don't have client_properties in mqtt (which holds the connection_name that is used for amqp) 
 but i agree that the client_id is a better string to use.